R/build-favicons.R

```diff
@@ -23,15 +13,33 @@ copy_favicons <- function(pkg = ".") {
 #' during package checking.
 #'
 #' @inheritParams as_pkgdown
+#' @param clobber If `TRUE`, re-create favicons from package logo.
 #' @export
-build_favicon <- function(pkg = ".") {
+build_favicons <- function(pkg = ".", clobber = FALSE) {
   pkg <- as_pkgdown(pkg)
 
+  rule("Building favicons")
+
   logo_path <- find_logo(pkg$src_path)
+
   if (is.null(logo_path)) {
-    stop("Can't find package logo.", call. = FALSE)
+    message("Can't find package logo PNG or SVG to build favicons.")
```

Comment:
Converting these errors to messages feels very dangerous to me. I think you'd be better off allowing `build_favicons()` to throw errors, and catching them in `init_site()`

Reply:
If this is an error, the implication is that we *require* packages to have a logo (i.e., it's an error if they don't). Obviously it's not a requirement to have a logo, so wouldn't a simple message be better?

Reply:
If you're calling a function called `build_favicons()`, and it fails to build favicons (for whatever reason), I think it should fail.

OTOH, the primary purpose of `init_site()` is not to create favicons, so it seems like it would be reasonable to not fail there.